src/ty/path.rs

```diff
@@ -63,16 +63,50 @@ impl IntoCompact for Path {
 
 impl Path {
 	/// Start building a Path with PathBuilder
-	#[cfg_attr(feature = "cargo-clippy", allow(clippy::new_ret_no_self))]
-	pub fn new() -> PathBuilder {
-		PathBuilder::new()
+	pub fn new(module_path: <MetaForm as Form>::String, ident: <MetaForm as Form>::String) -> Result<Path, PathError>
```

Comment:
I think it makes sense to switch parameters because `module_path` might be a longer expression in general.